diff --git a/ui/v2.5/graphql/stash-tv.graphql b/ui/v2.5/graphql/stash-tv.graphql
new file mode 100644
index 00000000..ac806bbb
--- /dev/null
+++ b/ui/v2.5/graphql/stash-tv.graphql
@@ -0,0 +1,53 @@
+query FindFullScenes(
+  $filter: FindFilterType
+  $scene_filter: SceneFilterType
+  $ids: [ID!]
+) {
+  findScenes(
+    filter: $filter
+    scene_filter: $scene_filter
+    ids: $ids
+  ) {
+    count
+    filesize
+    duration
+    scenes {
+      ...SceneData
+    }
+  }
+}
+
+query FindSceneMarkersForTv(
+  $filter: FindFilterType
+  $scene_marker_filter: SceneMarkerFilterType
+  $ids: [ID!]
+) {
+  findSceneMarkers(
+    scene_marker_filter: $scene_marker_filter
+    filter: $filter
+    ids: $ids
+  ) {
+    count
+    scene_markers {
+      ...SceneMarkerData
+      scene {
+        ...SceneData
+      }
+    }
+  }
+}
+
+query GetStashConfigForTv {
+  configuration {
+    plugins
+    ui
+  }
+  availableSavedSceneFilters: findSavedFilters(mode: SCENES) {
+    id
+    name
+  }
+  availableSavedMarkerFilters: findSavedFilters(mode: SCENE_MARKERS) {
+    id
+    name
+  }
+}
